third_party/txt/src/txt/paragraph.cc

```diff
@@ -559,8 +580,18 @@ void Paragraph::Layout(double width, bool force) {
     double justify_x_offset = 0;
     std::vector<PaintRecord> paint_records;
 
+    size_t run_index = -1;
     for (auto line_run_it = line_runs.begin(); line_run_it != line_runs.end();
          ++line_run_it) {
+      // Determine if the current run is a "ghost run" (does not impact
+      // centering/layout/etc). See above declaration of 'ghost_run_count' for
+      // more details.
+      run_index++;
+      bool is_ghost_run = false;
```

Comment:
Would it make sense to add the `is_ghost` flag to the `BidiRun` class?

Reply:
So a bidi run can actually be split into multiple parts, where the beginning half is a normal run and the last half is span of trailing whitespaces. Only once we split it into `line_runs` do we reach the level where a 'ghost' run can be defined.

For example, a line_range can have the range [0, 20, 40) where [20, 40) are whitespace. A corresponding bidi run can then be [0, 40), which should be broken into [0, 20) and [20, 40) where the first run is normal and the second is a 'ghost'.

Reply:
`line_runs` is a list of `BidiRun` instances.  I was thinking that it would be simpler to add a flag to the `BidiRun` struct to track the ghost runs instead of keeping state about which `line_runs` indexes are ghost runs.

Or maybe a better option would be to create a `LineRun` struct that includes all the fields of `BidiRun` plus the ghost flag.